### ISSUE TITLE
Argument of type `bool` evaluates all input to `True`

### ISSUE THREAD

**Issue by benghaziboy** [flask-restful/flask-restful#397]
```
class TestAPI(restful.Resource):

    reg_parser = reqparse.RequestParser()
    reg_parser.add_argument('test_bool', type=bool, location='form')

    def post(self):
        args = self.reg_parser.parse_args()

        return {
            'test_bool': args['test_bool']
            }
```

```
class TestTestCase(ApiTestCase):

    def test_bool_reg_parser(self):
        response = self.app.post(url_for('test_parser'), data={'test_bool': False})
        data = json.loads(response.data)


        assert data['test_bool'] == False
```

```
   Traceback (most recent call last):
    app/users/tests.py line 24 in test_bool_reg_parser
      assert data['test_bool'] == False
   AssertionError: 

```

```
print data
{u'test_bool': True}
```

**Comment by joshfriend** [comment]
This is because you can't use the python `bool` type to cast strings to boolean:

```python
>>> bool('true')
True
>>> bool('false')
False
```

Python only considers a string "Falsy" if it is an empty string, all other strings are `True`:

```python
>>> bool('')
False
```

TL;DR use `flask_restful.inputs.boolean` instead

**Comment by antonioromero** [comment]
Thanks @joshfriend can you update this issue in docs? It would be pretty useful.

**Comment by ibushong** [comment]
> This is because you can't use the python `bool` type to cast strings to boolean:
> 
> ```python
> >>> bool('true')
> True
> >>> bool('false')
> False
> ```

@joshfriend I think you mean `bool('false')` is `True`...
